### PR TITLE
feat: table remote sorting feature

### PR DESCRIPTION
**Pull request by sagarnaikjuspay**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
table remote filter feature
image

## How did you test it?
default no sorting will be applied, when u click the icons, the respective column of the table should sort asc or desc order, and u can apply only one sort at the column at a time, no multiple colmn sorting supoort at same time

## Where to test it?
- [x] INTEG
- [x] SANDBOX
- [x] PROD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/Order/Orders.res [hunk 1]
```diff
@@ -34,6 +37,16 @@ let make = (~previewOnly=false) => {
...
 
+        let sortObj = sortAtomValue->Dict.get("Orders")->Option.getOr(defaultSort)
+        if sortObj.sortKey->LogicUtils.isNonEmptyString {
+          let orderObj =
+            [
+              ("on", sortObj.sortKey->JSON.Encode.string),
+              ("by", sortObj->OrderTypes.getSortString->JSON.Encode.string),
+            ]->Dict.fromArray
+          filters->Dict.set("order", orderObj->JSON.Encode.object)
+        }
```
Comment (PritishBudhiraja) [inline src/screens/Order/Orders.res:48]:
```suggestion
        let sortObj = sortAtomValue->Dict.get("Orders")->Option.getOr(defaultSort)
        if sortObj.sortKey->isNonEmptyString {
          filters->Dict.set(
            "order",
            [
              ("on", sortObj.sortKey->JSON.Encode.string),
              ("by", sortObj->OrderTypes.getSortString->JSON.Encode.string),
            ]->getJsonFromArrayOfJson,
          )
        }
```

File: src/screens/Order/Orders.res [hunk 2]
```diff
@@ -77,9 +90,18 @@ let make = (~previewOnly=false) => {
...
     None
   }, (offset, filters, searchText))
 
+  React.useEffect(() => {
+    Some(
+      () => {
+        setSortAtom(_ => [("Orders", defaultSort)]->Dict.fromArray)
+      },
+    )
+  }, [])
```
Comment (PritishBudhiraja) [inline src/screens/Order/Orders.res:103]:
```suggestion
  React.useEffect(() => {
    Some(() => setSortAtom(_ => [("Orders", defaultSort)]->Dict.fromArray))
  }, [])
```

File: src/screens/HSwitchRemoteFilter.res [hunk 3]
```diff
@@ -203,6 +204,21 @@ module RemoteTableFilters = {
       None
     }, [filterValue])
 
+    let (dict, _) = Recoil.useRecoilState(LoadedTable.sortAtom)
```
Comment (PritishBudhiraja) [inline src/screens/HSwitchRemoteFilter.res:207]:
```suggestion
    let dict = Recoil.useRecoilValueFromAtom(LoadedTable.sortAtom)
```

File: src/components/LoadedTable.res [hunk 4]
```diff
@@ -518,12 +517,7 @@ let make = (
     None
   }, [filteredDataLength])
 
-  let filteredData = React.useMemo(() => {
```
Comment (JeevaRamu0104) [inline src/components/LoadedTable.res:521]: pass extra props and handle it don't remove the logic

File: src/screens/Order/Orders.res [hunk 5]
```diff
@@ -34,6 +37,19 @@ let make = (~previewOnly=false) => {
...
+        let sortObj = sortAtomValue->Dict.get("Orders")->Option.getOr(defaultSort)
+        if sortObj.sortKey->LogicUtils.isNonEmptyString {
+          filters->Dict.set(
+            "order",
+            [
+              ("on", sortObj.sortKey->JSON.Encode.string),
+              ("by", sortObj->OrderTypes.getSortString->JSON.Encode.string),
+            ]
+            ->Dict.fromArray
+            ->JSON.Encode.object,
```
Comment (Riddhiagrawal001) [inline src/screens/Order/Orders.res:49]: LogicUtils.getJsonFromArrayOfJson

File: src/components/Table.res [hunk 6]
```diff
@@ -443,12 +443,7 @@ module TableHeadingCell = {
                     let selfClass = "self-end"
                     <div className={`flex flex-row ${selfClass} items-center`}>
                       <SortAction
-                        item
-                        sortedObj
-                        setSortedObj
-                        sortIconSize
-                        filterRow
-                        isLastCol={isLastCol && !isFrozen && !isFirstCol}
```
Comment (Riddhiagrawal001) [inline src/components/Table.res:451]: Why this condition is changed ?

File: src/components/ReactWindowTable.res [hunk 7]
```diff
@@ -876,12 +837,7 @@ let make = (
     actualData
   }
 
-  let filteredData = React.useMemo(() => {
-    switch sortedObj {
-    | Some(obj: Table.sortedObject) => sortArray(actualData, obj.key, obj.order)
-    | None => actualData
-    }
-  }, (sortedObj, customGetObjects, actualData, getObjects))
+  let filteredData = actualData
```
Comment (Riddhiagrawal001) [inline src/components/ReactWindowTable.res:840]: Directly can use actualData

File: src/components/Table.res (hunk 6 above)
Comment (sagarnaikjuspay) [reply src/components/Table.res:451]: Before the filter is defaultly not supported in the last colomn which shoudn't be the case

File: src/screens/Order/Orders.res [hunk 8]
```diff
@@ -77,9 +93,14 @@ let make = (~previewOnly=false) => {
     if filters->isNonEmptyValue {
       fetchOrders()
     }
+
     None
   }, (offset, filters, searchText))
 
+  React.useEffect(() => {
+    Some(() => setSortAtom(_ => [("Orders", defaultSort)]->Dict.fromArray))
```
[the remark this answers is not in the document]
Comment (sagarnaikjuspay) [reply src/screens/Order/Orders.res:101]: `  React.useEffect(() => {
    Some(() => setSortAtom(_ => [("Orders", defaultSort)]->Dict.fromArray))
  }, [])`
  
  setting the sort obj atom value to default value when page exits

**Review by JeevaRamu0104 — DISMISSED** [review]

**Review by JeevaRamu0104 — APPROVED** [review]

**Review by gitanjli525 — APPROVED** [review]

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**